dace/frontend/python/replacements.py

```diff
@@ -4232,10 +4231,40 @@ def _ndarray_copy(pv: ProgramVisitor, sdfg: SDFG, state: SDFGState, arr: str) ->
 @oprepo.replaces_method('Array', 'fill')
 @oprepo.replaces_method('Scalar', 'fill')
 @oprepo.replaces_method('View', 'fill')
-def _ndarray_fill(pv: ProgramVisitor, sdfg: SDFG, state: SDFGState, arr: str, value: Number) -> str:
-    if not isinstance(value, (Number, np.bool_)):
-        raise mem_parser.DaceSyntaxError(pv, None, "Fill value {f} must be a number!".format(f=value))
-    return _elementwise(pv, sdfg, state, "lambda x: {}".format(value), arr, arr)
+def _ndarray_fill(pv: ProgramVisitor, sdfg: SDFG, state: SDFGState, arr: str, value: Union[str, Number,
+                                                                                           sp.Expr]) -> str:
+    assert arr in sdfg.arrays
+
+    if isinstance(value, sp.Expr):
+        raise NotImplementedError(
+            f"`{arr}.fill` is not implemented for symbolic expressions (`{value}`).")  # Look at `full`.
+
+    if isinstance(value, (Number, np.bool_)):
+        body = value
+        inputs = {}
+    elif isinstance(value, str) and value in sdfg.arrays:
+        value_array = sdfg.arrays[value]
+        if not isinstance(value_array, data.Scalar):
+            raise mem_parser.DaceSyntaxError(
+                pv, None, f"`{arr}.fill` requires a scalar argument, but `{type(value_array)}` was given.")
+        body = '__inp'
+        inputs = {'__inp': dace.Memlet(data=value, subset='0')}
+    else:
+        raise mem_parser.DaceSyntaxError(pv, None, f"Unsupported argument `{value}` for `{arr}.fill`.")
```

Comment:
```suggestion
        raise NotImplementedError(
            f"{arr}.fill is not implemented for symbolic expressions ({value}).")  # Look at `full`.

    if isinstance(value, (Number, np.bool_)):
        body = value
        inputs = {}
    elif isinstance(value, str) and value in sdfg.arrays:
        value_array = sdfg.arrays[value]
        if not isinstance(value_array, data.Scalar):
            raise mem_parser.DaceSyntaxError(
                pv, None, f"{arr}.fill requires a scalar argument, but {type(value_array)} was given.")
        body = '__inp'
        inputs = {'__inp': dace.Memlet(data=value, subset='0')}
    else:
        raise mem_parser.DaceSyntaxError(pv, None, f"Unsupported argument '{value}' for {arr}.fill.")
```

Reply:
I removed the backticks (`), since existing error reporting doesn't seem to use them.